### PR TITLE
[#9203] Replace trove4j with fastutils

### PR DESCRIPTION
**Pull request by FirstTimeInForever**
Replace the obsolete `trove4j` library with the `fastutil` as per #9203.

### CODE REVIEW

File: languagetool-core/pom.xml [hunk 1]
```diff
@@ -208,8 +208,8 @@
             <artifactId>annotations</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.jetbrains.intellij.deps</groupId>
-            <artifactId>trove4j</artifactId>
+            <groupId>it.unimi.dsi</groupId>
+            <artifactId>fastutil</artifactId>
```
Comment (donnerpeter) [inline languagetool-core/pom.xml:212]: Maybe use a stripped version from IJ?

Comment (FirstTimeInForever) [reply languagetool-core/pom.xml]: Oh, yes, I kind of forgot about it. But we decided to just use the official `fastutil-core`, as the IJ version does not worth the trouble to use outside of IJ.

Comment (donnerpeter) [reply languagetool-core/pom.xml]: I hope it's OK to add a 22 MB dependency

Comment (FirstTimeInForever) [reply languagetool-core/pom.xml]: `fastutil-core` is only `6.1MB`. See the commit below.

**Pull request merged**